src/Lib/Modes/PathEditMode.vala

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2021 Alecaddd (https://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Ashish Shevale <shevaleashish@gmail.com>
+*/
+
+public class Akira.Lib.Modes.PathEditMode : AbstractInteractionMode {
+
+    public weak Lib.ViewCanvas view_canvas { get; construct; }
+    public Lib.Items.ModelInstance instance { get; construct; }
+    public bool is_insert { get; construct; }
+    // keep track of previous point for calculating relative positions of points
```

Comment:
Little styling nit.
We're trying to write inline comments with this format:
`// Capital letter to start and then end with a period.`